compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1401,21 +1401,21 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     ///
     /// Given a function definition like:
     ///
-    /// ```rust
+    /// ```rust,ignore (pseudo-Rust)
```

Comment:
Can't you just leave these here as-is? They shouldn't complain because `doctest = false` is set (and the other PRs jyn mentioned already are working on enabling it).

Reply:
It does fail actually.

stdout:
```text
---- src/lib.rs - LoweringContext<'a,'hir>::lower_opaque_impl_trait (line 1404) stdout ----
error[E0404]: expected trait, found derive macro `Debug`
 --> src/lib.rs:1405:16
  |
3 | fn test<'a, T: Debug>(x: &'a T) -> impl Debug + 'a {
  |                ^^^^^ not a trait
  |
help: consider importing one of these items instead
  |
2 | use core::fmt::Debug;
  |
2 | use std::fmt::Debug;
  |

error[E0404]: expected trait, found derive macro `Debug`
 --> src/lib.rs:1405:41
  |
3 | fn test<'a, T: Debug>(x: &'a T) -> impl Debug + 'a {
  |                                         ^^^^^ not a trait
  |
help: consider importing one of these items instead
  |
2 | use core::fmt::Debug;
  |
2 | use std::fmt::Debug;
```

since this fits better, I updated it as:
```diff
diff --git a/compiler/rustc_ast_lowering/src/lib.rs b/compiler/rustc_ast_lowering/src/lib.rs
index 14f1a0f4019..5cb681b930b 100644
--- a/compiler/rustc_ast_lowering/src/lib.rs
+++ b/compiler/rustc_ast_lowering/src/lib.rs
@@ -1401,7 +1401,9 @@ fn lower_ty_direct(&mut self, t: &Ty, itctx: &ImplTraitContext) -> hir::Ty<'hir>
     ///
     /// Given a function definition like:
     ///
-    /// ```rust,ignore (pseudo-Rust)
+    /// ```rust
+    /// use std::fmt::Debug;
+    ///
     /// fn test<'a, T: Debug>(x: &'a T) -> impl Debug + 'a {
     ///     x
     /// }
```